portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,16 +37,23 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("postTime", SortDirection.DESCENDING);
+    int maxNoComments = getMaxComments(request);
```

Comment:
I'm assuming the "No" in `maxNoComments` is short for number. In general avoid abbreviations and shortening words because readers have to think a lot harder to understand what abbreviations mean then full words. In this case I think that you should just drop the "No", because `int maxComments` is enough to show that this variable is the maximum number of comments.

Reply:
```suggestion
    int maxComments = getMaxComments(request);
```